java/src/jmri/jmrit/dispatcher/AutoActiveTrain.java

```diff
@@ -707,6 +707,8 @@ protected synchronized void setupNewCurrentSignal(AllocatedSection as, boolean f
             Block nB = _nextBlock;
             if (as == null) {
                 as = _currentAllocatedSection;
+            }else{
```

Comment:
If the _currentAllocatedSection is null here then there is a serious problem elsewhere as you cannot set a signal until you have a currentallocated section. This routine should not be called unless there is a currentallocatedsection.

Reply:
There may be a serious problem elsewhere, but this resolved an exception which stopped trains running. Maybe I need to send you the exception that caused the problem. I do not know Dispatcher well enough to solve the exception than do this.

Reply:
When starting a train from outside the transit, it will allocate the starting section as the current section, if it cannot allocate it, it will be null. If you use the interface it will list the sections that maybe used to start the train from outside the transit, if you use one of those sections on the list there should be no error, if you programatically set a section not on the list then the error you see will occur. You must check the starting section is valid. The most likely causes for not being valid is that it is not connected directly to the transit with no intervening sections, or that the starting section cannot use the transit due to having to change the loco direction. If the cause is the former create a new section covering all the blocks to the meet with the transit, if it is the latter then there is no code for changing direction for reversing loco in mid transit. Create a train to get into the transit, and when terminated start a new train.

Reply:
Hi when setting the transits I do this automatically using the dispatcher system I have written, This always starts the train from a section outside the transit. The traininfo file must be corect in that if I use the Activate new train panel to setup and run the same train info file everything works OK. However from the dispatcher system I am using the createactivetrain routine, and setting the direction automatically. I will have a look to see whether a parameter is wrong. I thought they were all OK because on most occasions the method works OK, but not on some.

Reply:
I grab all the details for the createactivetrain routine from the traininfo file

Reply:
What do you mean by  'if it is the latter then there is no code for changing direction for reversing loco in mid transit'?  I only run the train once in one direction and do not change the direction of the train

Reply:
I have removed the edit, so hopefully the changes to dispatcher system can go through. i have had a look at the code and can see that the _currentAllocatedSection should be set here. It isn't and it is caused by the way in which I call the dispatch. If I call a dispatch and it works all dispatches subsequent to that work. It is only if I call one that doesn't work first that the problem occurs. Back to the drawing board.

Reply:
edit: can see that the _currentAllocatedSection should not be set here